public/app/plugins/datasource/loki/LogContextProvider.ts

```diff
@@ -248,6 +248,14 @@ export class LogContextProvider {
           }
         }
       }
+    } else if (query && isQueryWithParser(query.expr).parserCount === 0) {
+      // this case handles labels added from structured metadata, because the label is `fromParser` but the query does not contain a parser
+      const parsedLabels = contextFilters.filter((filter) => filter.fromParser && filter.enabled);
+      for (const parsedLabel of parsedLabels) {
+        if (parsedLabel.enabled) {
+          expr = addLabelToQuery(expr, parsedLabel.label, '=', parsedLabel.value, LabelType.StructuredMetadata);
+        }
+      }
```

Comment:
I think it make sense, but I am wondering, if we want to refactor this whole logic cause these both `ifs` are very similar, only difference is if we add parser or not.

Reply:
Also, maybe we should change `fromParser` prop to `indexed` and mark indexed labels instead, cause `fromParser` now has both - parsed and structured metadata labels, which is not intuitive. What do you think?

Reply:
> Also, maybe we should change fromParser prop to indexed and mark indexed labels instead, cause fromParser now has both - parsed and structured metadata labels, which is not intuitive. What do you think?

I think that's a good idea, I would still just change it to `nonIndexed` for now. So we don't introduce a logical change that might cause a bug.